docs/benchmarks/driving_smarts_2022.rst

```diff
@@ -0,0 +1,39 @@
+.. _driving_smarts_2022:
+
+Driving SMARTS 2022
+===================
+
+The Driving SMARTS 2022 is a benchmark derived from the
+NeurIPS 2022 Driving SMARTS Competition.
+
+This benchmark is intended to address the following requirements:
+
+-  The benchmark should use an up to date version of gym to simplify the
+   interface. (we used gymnasium)
+-  The competition gym environment should strictly follow the
+   requirements of the gym interface to allow for obvious actions and
+   observations.
+-  The observations out of this environment should be just data.
+-  The benchmark runner should be configurable for future benchmark
+   versions.
+-  Added benchmarks should be versioned.
+-  Added benchmarks should be discover-able.
+
+See `NeurIPS 2022 Driving SMARTS <https://smarts-project.github.io/archive/2022_nips_driving_smarts/>`_ page for historical context.
+
+This benchmark allows ego agents to use any one of the following action spaces.
+
++ :attr:`~smarts.core.controllers.ActionSpaceType.TargetPose`
++ :attr:`~smarts.core.controllers.ActionSpaceType.RelativeTargetPose`
```

Comment:
I do not see how we enforce this anymore. I think this requires a change to the environment constructor

https://github.com/huawei-noah/SMARTS/blob/7ff08fc7ef21d4bf389d6ef303793cec40368bf4/smarts/env/gymnasium/driving_smarts_competition_env.py#L57-L67

and how the environments are constructed by getting the agent spec to apply settings:

https://github.com/huawei-noah/SMARTS/blob/7ff08fc7ef21d4bf389d6ef303793cec40368bf4/smarts/benchmark/entrypoints/benchmark_runner_v0.py#L49-L59

Reply:
I have propagated `AgentInterface` through `benchmark_runner_v0` into the `driving_smarts_competition_v0_env`. This should allow action space to be configurable by the agents.

Considering an environment should just output one fixed type of observation for all agents, the `driving_smarts_competition_v0_env` was configured to have a fixed `observation_options=ObservationOptions.unformatted`.